subsys/usb/device_next/class/usbd_cdc_acm.c

```diff
@@ -202,6 +206,15 @@ static int usbd_cdc_acm_request(struct usbd_class_data *const c_data,
 	const struct device *dev = usbd_class_get_private(c_data);
 	struct cdc_acm_uart_data *data = dev->data;
 	struct udc_buf_info *bi;
+	const struct udc_api *udc_api = uds_ctx->dev->api;
+
+	if (IS_ENABLED(CONFIG_USBD_CDC_ACM_SHORT_PACKET_TERMINATE_ENABLED)) {
+		if ((data->ep_mps_mask == 0) && (udc_api->device_speed)) {
+			data->ep_mps_mask =
+				(udc_api->device_speed(uds_ctx->dev) == UDC_BUS_SPEED_FS) ? 0x3F
+											  : 0x1FF;
```

Comment:
Please familiarize yourself with the code base first.

Reply:
Hi @jfischer-no, 
Thanks for the pointers below, let me study the existing zlp handling and get back with a handling that matches the existing design.  Thanks again 😉

Reply:
Hi @jfischer-no, 
I have changed it to similar handling as what you have pointed out in CDC ECM. Please help to have a look. 😄